api/defaultHttpClient.go

```diff
@@ -0,0 +1,30 @@
+package api
+
+import (
+	"net/http"
+	"sync"
+	"time"
+)
+
+type httpClient struct {
+	client *http.Client
+}
+
+// NewDefaultHttpClient will create a default http client which can be queried to GET url responses
+func NewDefaultHttpClient(requestTimeoutSec uint64) *httpClient {
+	client := http.DefaultClient
+
+	var mutHttpClient sync.RWMutex
+	mutHttpClient.Lock()
+	client.Timeout = time.Duration(requestTimeoutSec) * time.Second
+	mutHttpClient.Unlock()
+
```

Comment:
please use `client := &http.Client{}` to create a new instance every time so we don't need to use the mutex for this

Reply:
As defined in `http` library, that's the same thing. Mutex was used to "replicate" the same behavior from our proxy, but after a discusstion with @bogdan-rosianu I found out that there was wrongfully used as well, so I've deleted it

> // DefaultClient is the default Client and is used by Get, Head, and Post.
var DefaultClient = &Client{}

Reply:
what i meant here is more for testing, if we use `http.DefaultClient` that's basically global; indeed we instantiate only one client, but we cannot run parallel unit tests for this, i think it's better to use a new instance of http client instead of using a global one

Reply:
This is basically a wrapper over http.Client to be easily mocked, so no real global object would be used in any test.
But your suggestion is solid, so I've applied the changes as suggested